oraclizeAPI_0.5.sol

```diff
@@ -845,36 +5826,181 @@ contract usingOraclize {
+    )
+        internal
+        pure
+        returns (byte _datasourceByte)
+    {
+        if (keccak256(abi.encodePacked(_datasourceString)) == keccak256('URL'))
+            return 0xFF;
+        if (keccak256(abi.encodePacked(_datasourceString)) == keccak256('Random'))
+            return 0xFE;
+        if (keccak256(abi.encodePacked(_datasourceString)) == keccak256('computation'))
+            return 0xFD;
+        if (keccak256(abi.encodePacked(_datasourceString)) == keccak256('WolframAlpha'))
+            return 0xFC;
+        if (keccak256(abi.encodePacked(_datasourceString)) == keccak256('IPFS'))
+            return 0xFB;
+        if (keccak256(abi.encodePacked(_datasourceString)) == keccak256('nested'))
+            return 0xFA;
+        if (keccak256(abi.encodePacked(_datasourceString)) == keccak256('Blockchain'))
+            return 0xF9;
+        if (keccak256(abi.encodePacked(_datasourceString)) == keccak256('swarm'))
+            return 0xF8;
+        return 0x00;
+    }
+
+    function oraclize_getRebroadcastCost(
+        uint256 _gasLimit,
+        uint256 _gasPrice
+    )
+        oraclizeAPI
+        internal
+        returns(uint256 _rebroadcastCost)
+    {
+        return oraclize.getRebroadcastCost(_gasLimit, _gasPrice);
+    }
+
+    function oraclize_requestCallbackRebroadcast(
+        bytes32 _queryId,
+        uint256 _gasLimit,
+        uint256 _gasPrice,
+        uint256 _queryPrice
+    )
+        oraclizeAPI
+        internal
+    {
+        return oraclize
+            .requestCallbackRebroadcast
+            .value(_queryPrice)
+            (
+                _queryId,
+                _gasLimit,
+                _gasPrice
+            );
+    }
+
+    function oraclize_setCustomTokenPayment(address _tokenAddress)
+        oraclizeAPI
+        internal
+    {
+        return oraclize.setCustomTokenPayment(_tokenAddress);
+    }
+
+    function oraclize_approveTokenAllowance(
+        address _tokenAddress,
+        uint256 _tokenAmount
+    )
+        oraclizeAPI
+        internal
+    {
+        ERC20Interface(_tokenAddress)
+            .approve(address(oraclize), _tokenAmount);
+    }
+
+    function oraclize_setAndApproveCustomTokenPayment(
+        address _tokenAddress,
+        uint256 _tokenAmount
+    )
+       oraclizeAPI
+       internal
+    {
+        oraclize_setCustomTokenPayment(_tokenAddress);
+        oraclize_approveTokenAllowance(_tokenAddress, _tokenAmount);
+    }
+
+    function oraclize_unsetAndRevokeCustomTokenPayment(address _tokenAddress)
+        oraclizeAPI
+        internal
+    {
+        oraclize_unsetCustomTokenPayment();
+        oraclize_approveTokenAllowance(_tokenAddress, 0);
+    }
+
+    function oraclize_unsetCustomTokenPayment()
+        oraclizeAPI
+        internal
+    {
+        return oraclize.unsetCustomTokenPayment();
+    }
+    /**
+     *
+     * @notice Oraclize helper functions follow...
+     *
+     */
+    function parseAddr(string memory _a)

```

Comment:
Where this function is used internally?

Reply:
It's not, it's a helper method for end contracts to use should they need. Internal just means the non-using contracts who inherit this don't have it added to their bytecode.